Plutarch/Builtin.hs

```diff
@@ -207,102 +184,40 @@ instance PIsData PByteString where
   pfromData x = pasByteStr # pforgetData x
   pdata x = punsafeBuiltin PLC.BData # x
 
+{- |
+  Instance for PBool following the Plutus IsData repr
+  given by @makeIsDataIndexed ''Bool [('False,0),('True,1)]@,
+  which is used in 'TxInfo' via 'Closure'.
+-}
+instance PIsData PBool where
+  pfromData x =
+    (phoistAcyclic $ plam $ \d -> toBool #$ tag # d)
```

Comment:
I'm not sure if @L-as already has optimizations for this, but if not - it may be beneficial to use a more direct application here-
```hs
phoistAcyclic $ plam $ \d -> pfstBuiltin # (pasConstr # d) #== 1
```
Same for the other method impl as well.

Reply:
This is indeed better.

Reply:
Not 100% sure what you meant for the other method but should be better now